src/algorithms/libs/beidou_b1c_signal_processing.cc

```diff
@@ -0,0 +1,859 @@
+    auto* b1c_data_primary_code_chips = static_cast<int32_t*>(volk_gnsssdr_malloc(static_cast<uint32_t>(BEIDOU_B1Cd_CODE_LENGTH_CHIPS) * sizeof(int32_t), volk_gnsssdr_get_alignment()));
+    const int32_t _samplesPerChip =  12;
+
+    make_b1cd(gsl::span<int32_t>(b1c_data_primary_code_chips, static_cast<uint32_t>(BEIDOU_B1Cd_CODE_LENGTH_CHIPS)), _prn); //generate Beidou B1C code, 1 sample per chip
+
+    _codeLength = _samplesPerChip * BEIDOU_B1Cd_CODE_LENGTH_CHIPS;
+    std::unique_ptr<float> _signal_B1C_data{new float[_codeLength]};
+    gsl::span<float> _signal_B1C_data_span(_signal_B1C_data, _codeLength);
+
+    beidou_b1cd_gen_float_11(_signal_B1C_data_span, gsl::span<int>(b1c_data_primary_code_chips, static_cast<uint32_t>(BEIDOU_B1Cd_CODE_LENGTH_CHIPS)));  // generate cboc 12 samples per chip
+    
+    volk_gnsssdr_free(b1c_data_primary_code_chips);
+}
+
+
+void beidou_b1c_code_gen_complex_sampled_boc_11(gsl::span<std::complex<float>> _dest,uint32_t _prn, int32_t _fs)
+{
+    //uint32_t _code_length =12 * BEIDOU_B1Cd_CODE_LENGTH_CHIPS;
+    //int32_t _code[_code_length];
+    
+    int32_t _samplesPerCode, _codeValueIndex;
+    float _ts;
+    float _tc;
+    const int32_t _codeLength =12 * BEIDOU_B1Cd_CODE_LENGTH_CHIPS;
+
+    //--- Find number of samples per spreading code ----------------------------
+    _samplesPerCode = static_cast<int>(static_cast<double>(_fs) / (static_cast<double>(BEIDOU_B1Cd_CODE_RATE_HZ) / static_cast<double>(_codeLength)));
+    
+    auto* real_code = static_cast<float*>(volk_gnsssdr_malloc(_samplesPerCode * sizeof(float), volk_gnsssdr_get_alignment()));
+    gsl::span<float> real_code_span(real_code, _samplesPerCode);
+    beidou_b1cd_code_gen_float_sampled_boc_11(real_code_span, _prn);
+    
+     //--- Find time constants --------------------------------------------------
+    _ts = 1.0 / static_cast<float>(_fs);                       // Sampling period in sec
+    _tc = 1.0 / static_cast<float>(BEIDOU_B1Cd_CODE_RATE_HZ);  // code chip period in sec
+    
+        
+    for (uint32_t i = 0; i < _samplesPerCode; i++)
+        {
+            //=== Digitizing =======================================================
+
+            //--- Make index array to read B1C code values -------------------------
+            _codeValueIndex = ceil((_ts * (static_cast<float>(i) + 1)) / _tc) - 1;
+
+            //--- Make the digitized version of the B1Cd code -----------------------
+            if (i == _samplesPerCode - 1)
+                {
+                    //--- Correct the last index (due to number rounding issues) -----------
+                    _dest[i] = std::complex<float>(1.0 - 2.0 * real_code_span[_codeLength - 1], 0);
+                }
+            else
+                {
+                    _dest[i] = std::complex<float>(1.0 - 2.0 * real_code_span[_codeValueIndex], 0);  //repeat the chip -> upsample
+                }
+        }           
+    volk_gnsssdr_free(real_code);
```

Comment:
Same comment as before regarding this type of memory allocation. Keep the consistency you developed before